src/Controller/Site.php

```diff
@@ -132,59 +103,43 @@ public function sitemap(Version $version)
     }
 
     /**
-     * Controller for sitemap-sitemap (multiple)
+     * Controller for sitemap-sitemap (multiple).
+     *
+     * @param Documentation $documentation
      *
-     * @return mixed
+     * @return Response
      */
-    public function sitemap_list()
+    public function getSitemapList(Documentation $documentation): Response
     {
         $twigVars = [
-            'versions' => $this->app['documentation']->getVersions(),
+            'versions' => $documentation->getVersions(),
         ];
 
         $xml = $this->render('sitemap_list.twig', $twigVars);
 
         return new Response($xml, 200, ['Content-Type' => 'application/xml']);
     }
 
-    protected function renderPage(Version $version, Page $page)
-    {
-        $twigVars = [
-            'page'            => $page,
-            'title'           => $page->getTitle(),
-            'version'         => $version,
-            'versions'        => $this->app['documentation']->getVersions(),
-            'default_version' => $this->app['documentation']->getDefault(),
-            'filelist'        => $this->app['documentation']->getFileStructure(),
-        ];
-
-        return $this->render($page['template'] ?: 'index.twig', $twigVars);
-    }
-
-    protected function render($template, array $variables = [])
-    {
-        return $this->app['twig']->render($template, $variables);
-    }
-
     /**
      * Controller for error pages.
      *
-     * @param \Exception $e
-     * @param Request    $request
-     * @param integer    $code
+     * @param \Exception    $e
+     * @param Request       $request
+     * @param Documentation $documentation
+     * @param int           $code
      *
      * @return Response|null
      */
-    public function error(\Exception $e, Request $request, $code)
+    public function error(\Exception $e, Request $request, Documentation $documentation, $code)
     {
-        $requestUri = explode('/', $request->getRequestUri());
+        $requestUri = \explode('/', $request->getRequestUri());
 
         // Don't trap Symfony shizzle.
-        if (in_array($requestUri[1], ['a', '_profiler']) || $this->app['debug']) {
+        if (\in_array($requestUri[1], ['a', '_profiler'], true) || $this->app['debug']) {
```

Comment:
Missed `$this->app['debug']`

Reply:
Yeah, knew about that one … was coming back for it this morning :+1:

Reply:
… had to implement the 404 handling still